### ISSUE TITLE
Search dialog: Hex to Text conversion of the search string is broken.

### ISSUE THREAD

**Issue by MKadaner** [FarGroup/FarManager#663]
### Far Manager version

3.0.6120.0

### OS version

10.0.22621.1413

### Steps to reproduce

1. Open _Find file_ dialog or _Search_ dialog in the internal Viewer.
2. Type `a b` in the search string.
3. Switch to Hex mode. Note that the search string looks as `61 20 62`.
4. Edit the search string and replace `20` with `00`.
5. Switch back to Text mode.

### Expected behavior

Search string is `a b` with the second character being U+0000. The zero character can be copied and pasted to the internal Editor, for example.

### Actual behavior

Search string consists of the single charcter `a`.

**Comment by MKadaner** [comment]
Please assign to me.

**Comment by alabuzhev** [comment]
> Edit the search string and replace 20 with 00.
> Switch back to Text mode.

This is by design and it was like this way before your changes.

Briefly, Far edit controls do support `\0`, but APIs like `DM_GETCONSTTEXTPTR` & `DM_SETTEXTPTR` don't for obvious reasons.
When you convert `61 00 62` back to text **using the ANSI code page** it becomes `a\0b` and gets chopped when sent to the dialog.

The workaround is simple: select `1200 | UTF-16 (Little endian)` from the code pages list if you want to search for UTF-16 LE data. No `\0`, no problem.

In other words, it is not about Hex/Text conversions per se, but about the API usage, or even about introducing better APIs (e.g. `DM_GETTEXT` will work, but using it within Far would be idiotic).

**Comment by MKadaner** [comment]
Re using UTF-16, please see [Far support forum](https://forum.farmanager.com/viewtopic.php?p=174050#p174050).

Interestingly, it worked in Viewer before the unifications. I suspect that it's not about API, but about Hex <-> Text conversion used in this place. I have not had time to look at the code right now, but I vaguely remember that the conversion deliberately stops at the zero byte. Let me check it later.

**Comment by alabuzhev** [comment]
Indeed, it worked in Viewer, the code below supposedly intentionally covers this case:
https://github.com/FarGroup/FarManager/blob/d7347f1e51a82c8e72030a42fea58ab8d7a88811/far/viewer.cpp#L2616-L2622

I was a bit too pessimistic, it's relatively easy to achieve without adding anything or resorting to too ugly methods:
- DM_GETTEXT populates item.PtrLength with the string size
- DM_GETCONSTTEXTPTR returns the data pointer
- The whole string thus can be extracted into Src
- Src is transformed into strTo 
- item.PtrLength and item.PtrData get populated with strTo size and data pointer
- DM_SETTEXT sets it without truncation.

It should be trivial to use the same approach in GetSearchReplaceString and Find File.

We don't even need to make this copy now because ConvertHexString accepts a string_view:
https://github.com/FarGroup/FarManager/blob/d7347f1e51a82c8e72030a42fea58ab8d7a88811/far/viewer.cpp#L2618

**Comment by alabuzhev** [comment]
Anyway, 6121.

**Comment by MKadaner** [comment]
Neat. Thank you!
